src/SonarScanner.MSBuild.PreProcessor/AnalysisConfigProcessing/Processors/TruststorePropertiesProcessor.cs

```
@@ -24,19 +24,75 @@ namespace SonarScanner.MSBuild.PreProcessor.AnalysisConfigProcessing.Processors;
+        var truststorePassword = PropertyValueOrDefault(SonarProperties.TruststorePassword, LocalSettings.TruststorePassword);
+
+        if (truststorePath is null)
+        {
+            if (operatingSystemProvider.IsUnix())
+            {
+                truststorePath = FindJavaTruststorePath();
+                truststorePassword ??= TruststorePasswordFromEnvironment();
+            }
+            else
+            {
+                MapProperty(config, JAVAX_NET_SSL_TRUST_STORE_TYPE, "Windows-ROOT");
+            }
+        }
+
+        MapProperty(config, JAVAX_NET_SSL_TRUST_STORE, truststorePath, ConvertToJavaPath, EnsureSurroundedByQuotes);
         config.LocalSettings.RemoveAll(x => x.Id is SonarProperties.TruststorePath or SonarProperties.TruststorePassword);
-        config.LocalSettings.Add(new Property(SonarProperties.TruststorePassword, password));
+
+        if (truststorePassword is not null)
+        {
+            MapProperty(config, JAVAX_NET_SSL_TRUST_STORE_PASSWORD, truststorePassword, EnsureSurroundedByQuotes);
+            config.LocalSettings.Add(new Property(SonarProperties.TruststorePassword, truststorePassword));
+        }
+    }
+
+    private static string TruststorePasswordFromEnvironment()
+    {
+        var sonarScannerOpts = Environment.GetEnvironmentVariable("SONAR_SCANNER_OPTS");
+        if (sonarScannerOpts is null)
+        {
+            return null;
+        }
+
+        return sonarScannerOpts.Split(' ').FirstOrDefault(x => x.StartsWith($"-D{JAVAX_NET_SSL_TRUST_STORE_PASSWORD}=")) is { } truststorePassword
+            ? truststorePassword.Substring($"-D{JAVAX_NET_SSL_TRUST_STORE_PASSWORD}=".Length)
+            : null;
+    }
+
+    private string FindJavaTruststorePath()
+    {
+        var javaHome = Environment.GetEnvironmentVariable("JAVA_HOME");
+        if (javaHome is null)
+        {
+            // TODO: propagate the error?
```

I don't think so. A debug log should be enough.

---

It will not fail before the end step then.

---

If the JRE provisioning is enabled, `JAVA_HOME` does not have to be set. There might be cases where the provisioned JRE trusts the server anyhow. The JRE comes with a default set of CA roots. If we fail it here, it would break https servers with a proper purchased certificate or one from https://letsencrypt.org/ 
Or am I miss something here?

---

If this is from letsencrypt, it should be trusted without giving a store.
My understanding of purchased certificates is that you don't need to provide a truststore as they should be valid from some public CA.

I would be for failing the execution to follow the fail fast principle but I am not 100% sure that in this case there is no scenario where it would work.

---

The scenario is:
* JRE provisioning is enabled and executed
* JAVA_HOME is not set
* The server is providing a certificate from a CA root trusted by the provisioned JRE (e.g. Let's Encrypt, Digi Cert, etc.)
* The user doesn't specify anything (`truststorePath is null`)
* We are on Linux
* `FindJavaTruststorePath` is called

-> We would fail the begin step, even though everything would just work.

---

Can we safely assume that the `cacerts` of JAVA_HOME would be the same the one from the provisioned JRE?

If we take the same scenario with JAVA_HOME set, we will tell the provisioned JRE to use the local `cacerts` rather that the one download. I wonder if there is a possibility for the scenario to succeed with one and not the other where in theory it should succeed with both.

---

The root ca certs change on a regular basis (see e.g. here https://learn.microsoft.com/en-us/security/trusted-root/release-notes). It is likely that we will have different CA roots in both locations.

---

Note: I think this is a problem the scanner-cli may have as well.

---

Also: As a user, you can install custom root CAs in you JAVA_HOME, but you can not install root CAs in the provisioned JRE. Therefore, we should prefer JAVA_HOME over the provisioned JRE.

---

Sounds good to me, I will only do a log.